From 70a8b71fd9d4951e69d1c4116dc28d2e1873d0eb Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 3 Aug 2016 14:30:19 +0200
Subject: [PATCH] logrotate.cron: use /var/lib/logrotate/logrotate.status as
 state file

Signed-off-by: rpm-build <rpm-build>

Patch re-generated by Junde Yhi to fit new codes.

--- a/examples/logrotate.cron	2017-06-22 00:50:27.190872698 +0800
+++ b/examples/logrotate.cron	2017-06-22 00:40:03.995901177 +0800
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-/usr/sbin/logrotate /etc/logrotate.conf
+/usr/sbin/logrotate -s /var/lib/logrotate/logrotate.status /etc/logrotate.conf
 EXITVALUE=$?
 if [ $EXITVALUE != 0 ]; then
     /usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
--- a/logrotate.8	2017-05-02 19:16:30.000000000 +0800
+++ b/logrotate.8	2017-06-22 00:49:23.088570890 +0800
@@ -78,6 +78,9 @@
 Tells \fBlogrotate\fR to use an alternate state file.  This is useful
 if logrotate is being run as a different user for various sets of
 log files.  The default state file is \fI/var/lib/logrotate.status\fR.
+Note that logrotate cron job on AOSC OS overrides the default with
+\fI/var/lib/logrotate/logrotate.status\fR in order to support read-only
+mounted root file systems.
 
 .TP
 \fB\-\-usage\fR
@@ -613,6 +616,7 @@
 tab(:);
 left l l.
 \fI/var/lib/logrotate.status\fR:Default state file.
+\fI/var/lib/logrotate/logrotate.status\fR:Default state file used by logrotate cron job on AOSC OS.
 \fI/etc/logrotate.conf\fR:Configuration options.
 .TE
 
